src/toInputObjectType.js

```diff
@@ -44,14 +45,15 @@ export type toInputObjectTypeOpts = {
 };
 
 export function toInputObjectType(
-  typeComposer: TypeComposer,
+  typeComposer: TypeComposer<any>,
   opts: toInputObjectTypeOpts = {},
   cache: Map<GraphQLObjectType, InputTypeComposer> = new Map()
 ): InputTypeComposer {
   if (typeComposer.hasInputTypeComposer()) {
     return typeComposer.getInputTypeComposer();
   }
 
+  const schemaComposer = typeComposer.constructor._schema;
```

Comment:
@nodkz reading this usage again - I have a small suggestion.

Perhaps a better API would expose on the TypeComposer (and other class that relies on `SchemaComposer`) a static `getSchemaComposer()` method that would provide the "parent" SchemaComposer.

(forgive me for the code mistakes - not a flow user..)
So, defining this interface
```
interface SchemaComposerDependent {
    getSchemaComposer() : SchemaComposer<*>
}
```

And then defining, for example, TypeMapper like so:
```
export class TypeMapper implements SchemaComposerDependent {
    getSchemaComposer() { throw new Error('TypeMapper can only be created by a SchemaComposer'); }
    ... rest of implementation
}
```

Then, when `SchemaComposer` generates dependent classes:
```
class TypeMapper extends _TypeMapper {
    getSchemaComposer() {
        return _schema;
    }
}
```

And finally from within TypeMapper somewhere:
```
set(name: string, type: GraphQLNamedType) {
    this.getSchemaComposer().set(name, type);
}
```
Which would throw if created directly.

I think in this case, there would be no need for the internal `_schema` field to exist and lexical scope of `getSchemaComposer` would capture the instance used. Also, the API is more explicit about the idea of "parent schema composer".

Reply:
Good additions. They will be added shortly.

Reply:
Added in such way https://github.com/graphql-compose/graphql-compose/pull/109/commits/1133f17a0b853154403ad737db1b26772ad7e7d4

Don't want to use inheritance, because in future `TypeComposer` may be inherited from `GraphQLObjectType`; `InputTypeComposer` from `GraphQLInputObjectType`. It will allow putting TC directly to vanilla GraphQL objects. It will be implemented somewhere in future.